src/doc/tutorials/PythonExpressions.rst

```diff
@@ -0,0 +1,280 @@
+A common practice is to make modifications to the script and test it using the Pseudocolor plot.
+Changes to the script can be made either by modifying the script in the *Python expression script* editor or modifying it in an external text editor and then reloading the script.
+Generaly speaking, modifying the script in the *Python expression script* editor is easier than doing it in an external text editor except that it is difficult to tell how many spaces are at the beginning of the line since the editor does not use a fixed width font.
+
+Developing the script in the *Python expression script* editor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following steps can be used to iteratively develop your script using the *Python expression script* editor.
+
+1. Edit the script.
+2. Click the *Apply* button.
+3. Go to *Variables->d* to change the variable to ``d``.
+   The first time you try your script this will not be necessary since the variable is already ``d``.
+4. Go to *Variables->MyExpression* to change the variable to "MyExpression" and execute the script.
+
+Developing the script in a text editor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
+
+The following steps can be used to iteratively develop your script using a text editor.
+
+Before you can modify the script you will need to save it.
+
+1. Click *Save script* to save the script.
+
+Now you are ready to modify the script.
+
+2. Edit the script with your favorit editor.
+3. Go to *Load script->File* to load the script.
+4. Click the *Apply* button.
+5. Go to *Variables->d* to change the variable to ``d``.
+   The first time you try your script this will not be necessary since the variable is already ``d``.
+6. Go to *Variables->MyExpression* to change the variable to "MyExpression" and execute the script.
+ 
+Python Expression Example 1
+---------------------------
+
+This example adds two cell centered variables.
+It demonstrates accessing multiple variables and performing simple operations with them to generate a result.
+
+Here is the example script.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 18-47
+
+Let us start off by creating a Pseudocolor plot from the expression.
+
+1. Copy the script into the *Python expression script* editor.
+2. Click the *Apply* button.
+3. Go to *Variables->MyExpression* to change the variable to the expression.
+
+.. figure:: images/PythonExpressions-Plot1.png
+
+   The Pseudocolor plot of MyExpression
+
+Now let us take a look at the script and see what each portion does.
+
+The ``__init__`` method provides information about the expression, including
+
+* That it inherits from ``SimplePythonExpression``.
+* The name of the expression.
+* A description of the expression.
+* A flag indicating that the output is not a point centered value.
+* A flag that the output is a scalar.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 19-24
```

Comment:
Regarding `:lines:` and the warning at top of the file, I guess that works ok. If we had CI support for this we could just add a dependency that would warn ~~committers~~ (whoops...only *hooks* work for committers) *pushers* that they might wanna make sure things still line up.

That said, Sphinx and reStructuredText are complex enough its easy to miss what is available in the way of functionality to help with this and so I just wanna ask, would this be any better...

```
.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
    :start-at: def __init__(self):
    :end-before: def modify_contract(self,contract):
```
That gets everything that might ever appear in the `__init__()` method regardless of line number.